src/compiler/checker.ts

```diff
@@ -18365,14 +18365,15 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         }
         if (accessNode) {
             const indexNode = getIndexNodeForAccessExpression(accessNode);
-            if (indexType.flags & (TypeFlags.StringLiteral | TypeFlags.NumberLiteral)) {
+            if (indexNode.kind !== SyntaxKind.BigIntLiteral && indexType.flags & (TypeFlags.StringLiteral | TypeFlags.NumberLiteral)) {
                 error(indexNode, Diagnostics.Property_0_does_not_exist_on_type_1, "" + (indexType as StringLiteralType | NumberLiteralType).value, typeToString(objectType));
             }
             else if (indexType.flags & (TypeFlags.String | TypeFlags.Number)) {
                 error(indexNode, Diagnostics.Type_0_has_no_matching_index_signature_for_type_1, typeToString(objectType), typeToString(indexType));
             }
             else {
-                error(indexNode, Diagnostics.Type_0_cannot_be_used_as_an_index_type, typeToString(indexType));
+                const typeString = typeToString(indexNode.kind === SyntaxKind.BigIntLiteral ? getBaseTypeOfLiteralType(getContextFreeTypeOfExpression(indexNode)) : indexType);

```

Comment:
Why do you need `getContextFreeTypeOfExpression` here?

Reply:
It gets a type from `indexNode`. Without this call/if I used `getBaseTypeOfLiteralType` on `indexType`, then the type returned is `string` which isn't right

Reply:
This doesn't seem right; when I debug, the literal appears to have all of the info that would be needed for an error message:

![image](https://github.com/user-attachments/assets/3fc78f26-43b6-4a8a-a93b-8ddf4f497507)

Reply:
Yeah, it seems like `pseudoBigIntToString` would be sufficient:

```ts
            const indexNode = getIndexNodeForAccessExpression(accessNode);
            if (indexType.flags & (TypeFlags.StringLiteral | TypeFlags.NumberLiteral)) {
                error(indexNode, Diagnostics.Property_0_does_not_exist_on_type_1, "" + (indexType as StringLiteralType | NumberLiteralType).value, typeToString(objectType));
            }
            else if (indexType.flags & TypeFlags.BigIntLiteral) {
                error(indexNode, Diagnostics.Property_0_does_not_exist_on_type_1, pseudoBigIntToString((indexType as BigIntLiteralType).value), typeToString(objectType));
            }
            else if (indexType.flags & (TypeFlags.String | TypeFlags.Number)) {
                error(indexNode, Diagnostics.Type_0_has_no_matching_index_signature_for_type_1, typeToString(objectType), typeToString(indexType));
            }
            else {
                const typeString = typeToString(indexType);
                error(indexNode, Diagnostics.Type_0_cannot_be_used_as_an_index_type, typeString);
            }
```